### PR TITLE
[15.0][FIX] password_security: correctly look at previous password history

### PR DESCRIPTION
**Pull request by astirpe**
Fixes password history issue https://github.com/OCA/server-auth/issues/89 for V15.
This is the porting of the fix discussed in https://github.com/OCA/server-auth/pull/448 and includes the tests written by @amh-mw

### CODE REVIEW

File: password_security/models/res_users.py [hunk 1]
```diff
@@ -185,18 +185,20 @@ def _check_password_history(self, password):
...
         """
         crypt = self._crypt_context()
-        for rec_id in self:
-            recent_passes = rec_id.company_id.password_history
-            if recent_passes < 0:
-                recent_passes = rec_id.password_history_ids
+        for user in self:
+            password_history = user.company_id.password_history
+            if not password_history:  # disabled
+                recent_passes = self.env["res.users.pass.history"]
```
Comment (amh-mw) [inline password_security/models/res_users.py]: I personally prefer to see .browse() after the model reference, as I think it's a more obviously understandable idiom, but I understand that they are synonyms.

```python
recent_passes = self.env["res.users.pass.history"].browse()
```

File: password_security/tests/test_password_history.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        with self.assertRaises(UserError):
+            user.write({"password": "admin"})
+        with self.assertRaises(UserError):
+            user.write({"password": "admit"})
+        user.write({"password": "badminton"})
+        self.assertEqual(len(user.password_history_ids), 3)
+
+        user.company_id.password_history = 0
+        user.write({"password": "badminton"})
+        self.assertEqual(len(user.password_history_ids), 4)
```
Comment (amh-mw) [inline password_security/tests/test_password_history.py:45]: Add this to get the last piece of code coverage:

```python
        user.company_id.password_history = -1
        with self.assertRaises(UserError):
            user.write({"password": "admin"})
```

**Review by amh-mw — CHANGES_REQUESTED** [review]

File: password_security/tests/test_password_history.py (hunk 2 above)
Comment (astirpe) [reply password_security/tests/test_password_history.py:45]: Done, thank you!

**Review by amh-mw — APPROVED** [review]
LGTM @OCA/tools-maintainers

**Comment by amh-mw** [comment]
@ivantodorovich Mind reviewing this fix for #448?

**Review by thomaspaulb — APPROVED** [review]

**Pull request merged**